kyo-direct/shared/src/main/scala/kyo/internal/AsyncShift.scala

```diff
@@ -17,9 +18,162 @@ end asyncShiftLowPriorityImplicit1
 object asyncShift extends asyncShiftLowPriorityImplicit1:
 
     transparent inline given shiftedChunk[A]: ChunkAsyncShift[A] = new ChunkAsyncShift[A]
+    transparent inline given shiftedMaybe: MaybeAsyncShift       = new MaybeAsyncShift
+    transparent inline given shiftedResult: ResultAsyncShift     = new ResultAsyncShift
 
 end asyncShift
 
+class ResultAsyncShift(using Frame) extends AsyncShift[Result.type]:
+
+    def map[F[_], E, A](result: Result.type, monad: CpsMonad[F])(ma: Result[E, A])[B](f: A => F[B]): F[Result[E, B]] =
+        monad match
+            case _: KyoCpsMonad[?] => ma match
+                    case Result.Success(a) => f(a).map(Result.Success(_))
+                    case Result.Failure(e) => monad.pure(Result.Failure(e))
```

Comment:
it seems to be missing `Result.Panic`.

Reply:
Yes indeed. 

I think `Result.Partial` should be the default `Result`, but that's another subject. 
```scala
type Result[E, A] = Result.Failure[E] | Result.Successful[A]
type Cause[E, A]  = Result[E, A] | Panic | Interrupt
```

Reply:
It seems to still be missing the handling of `Panic`?